sway-lib-std/src/storage.sw

```diff
@@ -681,6 +681,262 @@ impl<V> StorageVec<V> {
+
+            i += 1;
+        }
+    }
+
+    /// Fills `self` with elements by cloning `value`.
+    ///
+    /// ### Arguments
+    ///
+    /// * value - Value to copy to each element of the vector.
+    ///
+    /// ### Storage Access
+    ///
+    /// * Reads: `1`
+    /// * Writes: `self.len()`
+    ///
+    /// ### Examples
+    ///
+    /// ```sway
+    /// storage {
+    ///     vec: StorageVec<u64> = StorageVec {},
+    /// }
+    ///
+    /// fn foo() {
+    ///     storage.vec.push(5);
+    ///     storage.vec.push(10);
+    ///     storage.vec.push(15);
+    ///     storage.vec.fill(20);
+    ///
+    ///     assert(20 == storage.vec.get(0).unwrap());
+    ///     assert(20 == storage.vec.get(1).unwrap());
+    ///     assert(20 == storage.vec.get(2).unwrap());
+    /// }
+    /// ```
+    #[storage(read, write)]
+    pub fn fill(self, value: V) {
+        let len = get::<u64>(__get_storage_key()).unwrap_or(0);
+
+        let mut i = 0;
+        while i < len {
+            store::<V>(sha256((i, __get_storage_key())), value);
+            i += 1;
+        }
+    }
+
+    /// Resizes `self` in place so that `len` is equal to `new_len`.
+    ///
+    /// If `new_len` is greater than `len`, `self` is extended by the difference, with each
+    /// additional slot being filled with `value`. If the `new_len` is less than `len`, `self` is
+    /// simply truncated.
+    ///
+    /// ### Arguments
+    ///
+    /// new_len - The new length to expand or truncate to
+    /// value - The value to fill into new slots if the `new_len` is greater than the current length
+    ///
+    /// ### Storage Access
+    ///
+    /// * Reads - `1`
+    /// * Writes - `if new_len > self.len() { new_len - len + 1 } else { 1 }`
+    ///
+    /// ### Examples
+    ///
+    /// ```sway
+    /// storage {
+    ///     vec: StorageVec<u64> = StorageVec {},
+    ///     vec2: StorageVec<u64> = StorageVec {},
+    /// }
+    ///
+    /// fn foo() {
+    ///     storage.vec.push(5);
+    ///     storage.vec.push(10);
+    ///     storage.vec.push(15);
+    ///     storage.vec.resize(4, 20);
+    ///
+    ///     assert(5 == storage.vec.get(0).unwrap());
+    ///     assert(10 == storage.vec.get(1).unwrap());
+    ///     assert(15 == storage.vec.get(2).unwrap());
+    ///     assert(20 == storage.vec.get(3).unwrap());
+    ///
+    ///     storage.vec2.push(5);
+    ///     storage.vec2.push(10);
+    ///     storage.vec2.push(15);
+    ///     storage.vec2.resize(2, 0);
+    ///
+    ///     assert(5 == storage.vec.get(0).unwrap());
+    ///     assert(10 == storage.vec.get(1).unwrap());
+    /// }
+    /// ```
+    #[storage(read, write)]
+    pub fn resize(self, new_len: u64, value: V) {
+        let len = get::<u64>(__get_storage_key()).unwrap_or(0);
+        if new_len > len {
+            let mut i = len;
+            while i < new_len {
+                store::<V>(sha256((i, __get_storage_key())), value);
+                i += 1;
```

Comment:
```suggestion
        let mut len = get::<u64>(__get_storage_key()).unwrap_or(0);
        if new_len > len {
            while len < new_len {
                store::<V>(sha256((len, __get_storage_key())), value);
                len += 1;
```

What about this? 1 fewer allocation? Is this cheaper to build / run?

Reply:
if we're doing that, we can also remove the `if` statement:

```rust
pub fn resize(self, new_len: u64, value: V) {
    let mut len = get::<u64>(__get_storage_key()).unwrap_or(0);
    while len < new_len {
        store::<V>(sha256((len, __get_storage_key())), value);
        len += 1;
    }
    store::<u64>(__get_storage_key(), new_len);
}
```